src/libraries/Microsoft.PowerFx.Connectors/Public/CdpRecordType.cs

```diff
@@ -102,14 +102,19 @@ private bool TryGetFieldType(string fieldName, bool ignorelationship, out Formul
 
             try
             {
+                if (TableResolver == null)
+                {
+                    throw new InvalidOperationException("TableResolver is not set.");
+                }
+
                 ConnectorType connectorType = TableResolver.ResolveTableAsync(tableName, CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
 
                 type = connectorType.FormulaType;
                 return true;
             }
             catch (Exception ex)
             {
-                TableResolver?.Logger.LogException(ex, $"Cannot resolve external table {tableName}");
+                TableResolver?.Logger?.LogException(ex, $"Cannot resolve external table {tableName}");
```

Comment:
Note it TableResolver is null, then we still won't log anything. 
And we still swallow.